packages/main/src/Toolbar.ts

```diff
@@ -0,0 +1,590 @@
+	 * @defaultvalue: "End"
+	 * @name sap.ui.webc.main.Toolbar.prototype.alignContent
+	 */
+	@property({ type: ToolbarAlign })
+	alignContent!: string;
+
+	/**
+	 * Actions, which will be displayed in the toolbar.
+	 * @type {Object}
+	 * @private
+	 */
+	@property({ type: Object, multiple: true })
+	actionsToBar!: Array<ToolbarItem>;
+	/**
+	 * Actions, that will be displayed inside overflow Popover.
+	 * @type {Object}
+	 * @private
+	 */
+	@property({ type: Object, multiple: true })
+	actionsToOverflow!: Array<ToolbarItem>;
+	/**
+	 * Cached actions width and the sum of all of them.
+	 * @type {Object}
+	 * @private
+	 */
+	@property({ type: Object })
+	ACTIONS_WIDTH_MAP!: ActionsWidthMap;
+	/**
+	 * Indicates the actions have been measured and the layout can be calculated.
+	 * @type {boolean}
+	 * @private
+	 */
+	@property({ type: Boolean })
+	actionsWidthMeasured!: boolean;
+	/**
+	 * Indicates the end of the resizing iteration.
+	 * @type {boolean}
+	 * @private
+	 */
+	@property({ type: Boolean })
+	resizing!: boolean;
+
+	/**
+	* Calculated width of the whole toolbar.
+	* @private
+	*/
+	@property({ type: Integer })
+	width!: number;
+
+	/**
+	* @private
+	* Calculated width of all the Toolbar items.
+	*/
+	@property({ type: Integer })
+	contentWidth!: number;
+
+	/**
+	 * Notifies the toolbar if it should show the items in a reverse way if Toolbar Popover needs to be placed on "Top" position.
+	 * @private
+	 * @type {sap.ui.webc.main.types.ToolbarAlign}
+	 */
+	@property({ type: Boolean })
+	reverseOverflow!: boolean;
+
+	/**
+ 	* Slotted Toolbar items
+	* @type {sap.ui.webc.main.IToolbarItem[]}
+	* @name sap.ui.webc.main.Toolbar.prototype.default
+	* @slot actions
+	* @public
+	*/
+	@slot({ "default": true, type: HTMLElement, invalidateOnChildChange: true })
+	actions!: Array<ToolbarItem>
+
+	_onResize!: ResizeObserverCallback;
+
+	constructor() {
+		super();
+
+		// actions displayed in the bar
+		this.actionsToBar = [];
+
+		// actions displayed in the toolbar overflow popover
+		this.actionsToOverflow = [];
+
+		// store for actions width
+		this.ACTIONS_WIDTH_MAP = {};
+
+		// indicates the actions have been measured and the layout can be calculated
+		this.actionsWidthMeasured = false;
+
+		// indicates if the bar is being reized at the moment
+		this.resizing = false;
+
+		// resize handler
+		this._onResize = this.onResize.bind(this);
+	}
+
+	static get OVERFLOW_BTN_SIZE(): number {
+		return 44;
+	}
+
+	static get ACTION_MARGIN() {
+		return 4;
+	}
+
+	static get PADDING() {
+		return 16;
+	}
+
+	/**
+	 * Lifecycle methods
+	 */
+	onEnterDOM() {
+		ResizeHandler.register(this, this._onResize);
+	}
+
+	onExitDOM() {
+		ResizeHandler.deregister(this, this._onResize);
+	}
+
+	async onAfterRendering() {
+		if (this.resizing) {
+			this.resizing = false;
+			return;
+		}
+
+		await renderFinished();
+		this.storeActionsWidth();
+		this.doLayout();
+	}
+
+	/**
+	 * Layout management
+	 */
+	doLayout(forceLayout = false) {
```

Comment:
I think this is too abstract as a name (although I know it's how this is called in OpenUI5) but if nothing better comes to mind, keep it